### PR TITLE
LP-2454 Write unit tests, cater missed edge cases and improve code

### PR DESCRIPTION
**Pull request by harris-irfan**

### CODE REVIEW

**Review by asadali145 — DISMISSED** [review]

**Comment by asadali145** [comment]
@harris-irfan We have some issue with an action related to the transifex translations, We need to check this before merging this one.

**Comment by harris-irfan** [comment]
> @harris-irfan We have some issue with an action related to the transifex translations, We need to check this before merging this one.

@asadali145 any next steps for me? The error states:

```
Error: Error: There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> harris/LP-2454_koa
```

**Comment by asadali145** [comment]
> > @harris-irfan We have some issue with an action related to the transifex translations, We need to check this before merging this one.
> 
> @asadali145 any next steps for me? The error states:
> 
> ```
> Error: Error: There is no tracking information for the current branch.
> Please specify which branch you want to merge with.
> See git-pull(1) for details.
> 
>     git pull <remote> <branch>
> 
> If you wish to set tracking information for this branch you can do so with:
> 
>     git branch --set-upstream-to=origin/<branch> harris/LP-2454_koa
> ```

@harris-irfan I think we can try what it suggested. Try this `git branch --set-upstream-to=origin/<branch> harris/LP-2454_koa` and push your branch again maybe. I don't have context of this issue, Maybe @MunamMubashir-arbisoft Can help with this action.

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1000/workflows/workflows_id/jobs/1000/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/mailchimp_pipeline/helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/constants.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/migrations/0004_auto_20210131_0952.py (0.0%): Missing lines 3,6,8,12
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/constants.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_inlines.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_forms.py (100%)
openedx/adg/lms/applications/tests/test_handlers.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
-------------
Total:   480 lines
Missing: 4 lines
Coverage: 99%
-------------
```

File: conf/locale/ar/LC_MESSAGES/djangojs.po [hunk 1]
```diff
@@ -9684,6 +9723,45 @@ msgstr ""
...
+"password reset email."
+msgstr ""
+
+#: themes/edx-platform/adg/lms/templates/student_account/password_reset.underscore
+msgid "SEND"
+msgstr ""
+
+#: themes/edx-platform/adg/lms/templates/student_account/register.underscore
+msgid "Already have an account?"
+msgstr "هل لديك حساب {platformName} مسبقاً؟"
```
Comment (asadali145) [inline conf/locale/ar/LC_MESSAGES/djangojs.po:9755]: @harris-irfan Please remove this `{platformName}` from the translated string. I think the translator has done this or I don't know where it came from.

**Review by asadali145 — DISMISSED** [review]
Just a translation fix. Apart from that, LGTM

**Review by asadali145 — APPROVED** [review]

File: conf/locale/ar/LC_MESSAGES/djangojs.po (hunk 1 above)
Comment (harris-irfan) [reply conf/locale/ar/LC_MESSAGES/djangojs.po:9755]: As discussed, we will be going to master with the current file, without incorporating the suggested change.

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1010/workflows/workflows_id/jobs/1010/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/mailchimp_pipeline/helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/constants.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/migrations/0004_auto_20210131_0952.py (0.0%): Missing lines 3,6,8,12
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/constants.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_inlines.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_forms.py (100%)
openedx/adg/lms/applications/tests/test_handlers.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
-------------
Total:   480 lines
Missing: 4 lines
Coverage: 99%
-------------
```

**Review by asadali145 — APPROVED** [review]

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1012/workflows/workflows_id/jobs/1012/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/mailchimp_pipeline/helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/constants.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/migrations/0004_auto_20210131_0952.py (0.0%): Missing lines 3,6,8,12
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/constants.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_inlines.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_forms.py (100%)
openedx/adg/lms/applications/tests/test_handlers.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
-------------
Total:   480 lines
Missing: 4 lines
Coverage: 99%
-------------
```
